Source/WebKit/UIProcess/Cocoa/WKStorageAccessAlert.mm

```diff
@@ -73,24 +73,37 @@ void presentStorageAccessAlertQuirk(WKWebView *webView, const WebCore::Registrab
 
     NSString *informativeText = [NSString stringWithFormat:WEB_UI_NSSTRING(@"This will allow “%@” and “%@” to track your activity.", @"Informative text for requesting cross-site cookie and website data access."), firstRequestingDomain.get(), secondRequestingDomain.get()];
     
-    displayStorageAccessAlert(webView, alertTitle, informativeText, nil, WTFMove(completionHandler));
+    displayStorageAccessAlert(webView, alertTitle, informativeText, WTFMove(completionHandler));
 }
 
 void presentStorageAccessAlertSSOQuirk(WKWebView *webView, const String& organizationName, const HashMap<WebCore::RegistrableDomain, Vector<WebCore::RegistrableDomain>>& domainPairings, CompletionHandler<void(bool)>&& completionHandler)
 {
-    NSString *alertTitle = [NSString stringWithFormat:WEB_UI_NSSTRING(@"Are you logging in to this website, and do you want to allow other %@ sites access to your website data while browsing the websites listed below?", @"Message for requesting cross-site cookie and website data access."), organizationName.createCFString().get()];
-    NSString *informativeText = [NSString stringWithFormat:WEB_UI_NSSTRING(@"Allowing access to website data is necessary for the website to work correctly.", @"Informative text for requesting cross-site cookie and website data access.")];
-
-    auto* accessoryTextList = [NSMutableArray arrayWithCapacity:domainPairings.size()];
-    for (const auto& domains : domainPairings) {
-        auto embeddedDomains = makeStringByJoining(domains.value.map([](auto& domain) { return domain.string(); }).span(), "\n  - "_s);
-        NSString *accessoryText = [NSString stringWithFormat:WEB_UI_NSSTRING(@"While you are visiting %@, the following related websites will gain access to their cookies:\n  - %@", @"Accessory text for requesting cross-site cookie and website data access."), domains.key.string().createCFString().get(), embeddedDomains.createCFString().get()];
-        [accessoryTextList addObject:accessoryText];
+    NSString *alertTitle = [NSString stringWithFormat:WEB_UI_NSSTRING(@"Do you want to allow %@ websites to access one another's cookies and website data?", @"Message for requesting cross-site cookie and website data access."), organizationName.createCFString().get()];
+
+    HashSet<WebCore::RegistrableDomain> allDomains;
```

Comment:
I think this could be `ListHashSet` for an ordered hash set, instead of creating a vector below.

Reply:
But then how would he sort them alphabetically later on, like he does now?

Reply:
Oh, yeah, I guess there isn't a good way to sort an ordered hash set. You would need to insert them alphabetically, so the way you have it now is probably best.